src/UI/Internal/SideBar.elm

```diff
@@ -52,14 +65,77 @@ desktopColumn cfg page menu =
         ]
 
 
-mobileDrawer :
+desktopNonPersistent : RenderConfig -> Element msg -> Menu msg -> Element msg
+desktopNonPersistent cfg page (Menu.Menu prop opt) =
+    let
+        alpha =
+            if prop.isExpanded then
+                0.5
+
+            else
+                0
+
+        overlay =
+            Element.el
+                (width fill
+                    :: height fill
+                    :: Element.alpha alpha
+                    :: (Palette.black
+                            |> Palette.toElementColor
+                            |> Background.color
+                       )
+                    :: css [ ( "transition", "opacity .2s" ) ]
+                )
+                Element.none
+    in
+    Element.row
+        [ width fill
+        , height fill
+        , viewSide cfg
+            { proportional = False, persistent = False }
+            (Menu.Menu { prop | isExpanded = True } opt)
+            |> sidebarTransitionContainer prop.isExpanded
+            |> Element.inFront
+        ]
+        [ slimHeaderView cfg
+            (prop.toggleMsg (not prop.isExpanded))
+            Palette.primary
+            |> Element.el [ paddingXY 14 24, Element.alignTop ]
+        , Element.el
+            [ width fill
+            , height fill
+            , Element.inFront overlay
+            ]
+            page
+        ]
+
+
+sidebarTransitionContainer : Bool -> Element msg -> Element msg
+sidebarTransitionContainer isExpanded =
+    let
+        transitionStyles =
+            if isExpanded then
+                [ ( "transition", "transform .4s" )
+                ]
+
+            else
+                [ ( "transform", "translate(-100%)" )
+                , ( "opacity", "0" )
+                , ( "pointer-events", "none" )
+                , ( "transition", "transform .4s, opacity .2s .4s" )
+                ]
+    in
+    Element.el (height fill :: css transitionStyles)
```

Comment:
if you don't like this approach we could start a discussion on animations/transitions because currently it seems like we haven't been doing anything on that end. I took a look at [elm-animator](https://package.elm-lang.org/packages/mdgriffith/elm-animator/latest/) and thought that it was waaay too much code for a simple transition, so this would be a simpler approach that trusts that the virtual dom will only patch the styles when the change happens. In some cases, like if a node changes from a `<div>` to a `<section>` between renders, it would not work but since we're using elm-ui **and** it's a pure language I don't see it happening.

Reply:
I don't think we had any animations until this point.
`elm-animator` looks like using a nuke to kill an ant...

By now, could we at least move all "unsafe CSS" to a separated module (`UI.Internal.Utils.Element` looks like it)? Then in the future, we could decide how to make them all safe and standardized. (Probably a call for the team-leader <@jouderianjr> to make) -- Also, talk with U{X,I}Team about what else should be animated...